src/quicktype-core/language/Swift.ts

```diff
@@ -952,6 +960,37 @@ ${this.accessLevel}class JSONAny: Codable {
         }
     };
 
+    private emitConvenienceMutator(c: ClassType, className: Name) {
+        this.emitLine(this.accessLevel, "func with(");
+        this.indent(() => {
+            this.forEachClassProperty(c, "none", (name, _, p, position) => {
+                this.emitLine(
+                    name,
+                    ": ",
+                    this.swiftPropertyType(p),
+                    "? = nil",
```

Comment:
@dvdsgl For an already optional argument this would add something like `:String?? = nil` is this intentional? If so, what would be the difference to `:String? = nil`? 
(I apologies if this seems out of the blue, discovered a sample today and wasn't sure what I'm seeing)

Reply:
I think that would be an bug.  Do you have a test case for that?  Could you open an issue?

Reply:
This is intentional. I'm open to other ways to specify nil for an optional property, but this is the most unadorned approach I could think of.

Reply:
Ah, I see - you can replace an optional value with `nil` by using `Optional.some(nil)`.  Ok, it's a feature, I guess :-)

Reply:
Or `Optional(nil)` — far from lovely but the simplest semantics I could think of…

It's still tricky so I'm open to improvements.

Reply:
I see. But isn't the output `with(test: String?? = nil)` technically invalid Swift and it should rather be `with(test: String? = nil)` which would be valid Swift syntax with the same result?

Reply:
It's valid Swift - it compiles in our test suite.

The functionality of `with` is to replace one or more property values with new values.  If you don't want to replace the value, you pass in `nil`.  But if the property is optional, `nil` is itself a valid value, so what if you want to replace the value with `nil`?  The ugly answer: If you don't want to replace it, you pass `nil`.  If you want to replace it with `nil`, you pass `Optional(nil)`.

Reply:
Thank you so much for the explanation! It makes total sense now!!